[DO NOT MERGE] Banks and Universities

joelAVS opened the pull request:
Added 2 new province buildings, banks and universities, and related events.

joelAVS committed:
New Buildings

Added 2 new buildings, banks and univerisites, and related events

joelAVS committed:
Flying university event moved

The Flying University event was moved to the banks and universities event file

joelAVS committed:
Bug fixes and terrain changes

Multiple bugfixes and now banks can be build in any terrain freely (Bank level needed is the same as railroad - 1)

joelAVS committed:
Fixed some bank terrain I forgot, changed bank and railway balance

joelAVS committed:
Bug fixes and balance changes

joelAVS committed:
Added Immigration submod support

Added support for the new buildings in the Direct Immigration submod

Also some adjusts and bugfixes

joelAVS committed:
Optimizations and the Flying Universities event moved back

Proper use of the OR expression and the Flying Universities and farmers bank events are back on it's original files, it was a mistake moving then to the new file.

joelAVS committed:
Abusive banking practices back to it's original file

Moved by mistake lol

joelAVS committed:
Small optimizations and adjusts

joelAVS committed:
Improved Direct Immigration mod UI

joelAVS committed:
More fixes for the Immigration submod

joelAVS committed:
Fixed small defect on the other province screen

joelAVS committed:
Terrain adjusted

Terrain restriction adjusted
Terrain-based MTTH removed, life rating is enough

joelAVS committed:
Final adjusts

joelAVS committed:
University MTTH adjusted

joelAVS committed:
Big Frame Implementation

joelAVS committed:
Multiple fixes for the new big frames

joelAVS committed:
2 new bank related crimes added

Greedy Bankers and Loan Sharks.

joelAVS committed:
Added banks and universities to provinces in south america

joelAVS committed:
Added universities to central and north america

joelAVS committed:
Added universities to canada

joelAVS committed:
Added banks around the world based on historical information

joelAVS committed:
Placed 67  universities around the world based on historical data

joelAVS committed:
Multiple fixes to the big frames

joelAVS committed:
Fixed crimes and added new localization

joelAVS committed:
Fixes for Other's Province screen with big frames

joelAVS committed:
Small fixes, the news UI is now TGC green

joelAVS committed:
news adjusted

joelAVS committed:
Region.txt problem idk what happened to it in this branch

joelAVS committed:
Balances changes to accommodate the new RGO changes

* Increased good demand into the late game
* Decreased RGO output tech buffs
* Anti Liquidity Crises Measures
  * Anti hoarding mechanic turned on by default
  * Tax Refunds
  * Increased workers salaries
  * Decreased gold miner banks

joelAVS committed:
Better money management

Money from AI building Banks and Universities now gets injected back into the economy

joelAVS committed:
Fixed extra bracked

Thanks validator :)

joelAVS committed:
Bug fixes and formatting

joelAVS committed:
Banks bug fix

joelAVS committed:
Small bug fixed

GeneralVonGelre commented on TGC/events/BanksAndUniversities.txt:
```diff
@@ -0,0 +1,406 @@
+        owner = { money = 10000
+                  NOT = { economic_policy = laissez_faire }
+				  ai = yes
+				  administration_spending = 0.6
```
Note that when checking for administration_spending the game will crash if it's done for a country that doesn't actually exist. Generally for safety reasons youd put an exists = yes check before doing this just to be absolutely sure.

It is also basically the only case where you'd ever use exists = yes funnily enough

GeneralVonGelre replied:
I'm not gonna comment this everytime you check for it, I'm sure you can figure that out on your own

GeneralVonGelre commented on TGC/events/BanksAndUniversities.txt:
```diff
@@ -0,0 +1,406 @@
+
+	trigger = {
+		has_building = bank
+		NOT = { administration_spending = 0.5 }
```
are you sure this works as a province-level trigger? I think it's a country scope in which case you gotta check this for the owner of the province rather than the province itself

joelAVS replied:
I had discussion with slav earlier, for some reason administration and education spending will work normally in a province event. I realized it shouldnt but it was working anyway later in development, and idk if I should ad the owner tag or not given the weird situation

joelAVS committed:
Exist fix and balance changes

joelAVS committed:
UI fix

Multiple small adjustments to the UI. Many elements centered and ajusted. Constrast of the flag shadow increased, big frame applied to post game screen.

joelAVS committed:
Formatting lobby.gui

joelAVS committed:
New building icon. Tax refund adjusted, added a bunch of owner tags..

joelAVS committed:
Formatting diplomacy.gui

Yes. I can't stand the extra break lines.

joelAVS committed:
Formatting again.

joelAVS committed:
Final fix

joelAVS committed:
Small localization fix.

joelAVS committed:
Need for clothes increased as everyday and luxury need.

This should fix the problematic cotton farming in early game.

joelAVS committed:
Cotton industry bandaid fix

joelAVS committed:
Religion Icon fix

joelAVS committed:
Submod names adjusted

Now the TGC main module always appears first

joelAVS committed:
Last minute university and bank nerf

joelAVS committed:
Be gone submod! (In this branch as well)
